roles/github/templates/workflow.yml.j2

```diff
@@ -1,3 +1,14 @@
+<%- macro format_file_name(file_name, is_title=false, is_subtitle=false) -%>
+  <%- set formatted_name = file_name | splitext | first -%>
+  <%- if is_title -%>
+    <%- set formatted_name = formatted_name | replace('-', ' ') | title | regex_replace('Vm','VM') -%>
+  <%- endif -%>
+  <%- if is_subtitle -%>
+    <%- set formatted_name = formatted_name | replace('-', ' ') | capitalize | regex_replace('vm','VM') -%>
+  <%- endif -%>
+  %%- formatted_name -%%
+<%- endmacro -%>
+
 <% if workflow.use_bespoke is defined and workflow.use_bespoke is true %>
   <%- extends workflow.file_name + ".j2" | string +%>
```

Comment:
Is this just intended to support templates within this repository, and not user-provided templates?

Reply:
If a user makes changes to the roles variables such as adding a new `workflow` it would benefit from the macro. However, not sure how to expose the macro to consumers of the collection such as a user bringing an entire template.

Reply:
I meant use_bespoke - should it possible to provide workflow templates from outside the role?

Reply:
It isn't looking likely. Issues such as https://github.com/ansible/ansible/issues/20442. Indicates changes to how Jinja2/Ansible templating functions which prevents templates from referencing templates from outside the scope of the role.

It will work if the template [task](https://github.com/stackhpc/stackhpc.kayobe_automation_workflows/blob/c6890c58a9b065ffb5023fec2230344ff7f9b2ec/roles/github/tasks/main.yml#L10) contains the following.

```
src: "{{ playbook_dir }}/templates/workflow.yml.j2"
```

However we need to edit the [workflow.yml.j2(https://github.com/stackhpc/stackhpc.kayobe_automation_workflows/blob/c6890c58a9b065ffb5023fec2230344ff7f9b2ec/roles/github/templates/workflow.yml.j2#L2) to some how load playbook templates if provided.

Reply:
In [kolla-ansible](https://opendev.org/openstack/kolla-ansible/src/branch/master/ansible/roles/common/templates/td-agent.conf.j2) I used `lookup('template', path)` to pull in custom config templates when they were outside the role. I don't know if we need to consider that here, but we should at least say whether it's currently possible in the README.

Reply:
Users can now provide templates from their `{{ playbook_dir }}/templates` directory. They must set `path_override` within the workflow definition.

```
github_build_kayobe_image:
  file_name: build-kayobe-docker-image.yml
  path_override: "{{ playbook_dir }}/templates/build-kayobe-docker-image.yml.j2"
```